### PR TITLE
MCOL-5227

### PR DESCRIPTION
**Pull request by mariadb-AndreyPiskunov**
Added tests to basic as well.

### CODE REVIEW

File: dbcon/joblist/jsonarrayagg.h [hunk 1]
```diff
@@ -0,0 +1,174 @@
+/* Copyright (C) 2014 InfiniDB, Inc.
```
Comment (drrtuy) [inline dbcon/joblist/jsonarrayagg.h]: 2014, really?

File: utils/rowgroup/rowaggregation.cpp [hunk 2]
```diff
@@ -3989,6 +4015,16 @@ void RowAggregationUM::setGroupConcatString()
...
       }
+
+      if (fFunctionCols[j]->fAggFunction == ROWAGG_JSON_ARRAY)
+      {
+        uint8_t* buff = data + fRow.getOffset(fFunctionCols[j]->fOutputColumnIndex);
+        uint8_t* gcString;
+        joblist::JsonArrayAggregatAgUM* gccAg = *((joblist::JsonArrayAggregatAgUM**)buff);
+        gcString = gccAg->getResult();
+        fRow.setStringField((char*)gcString, fFunctionCols[j]->fOutputColumnIndex);
+        // gccAg->getResult(buff);
```
Comment (drrtuy) [inline utils/rowgroup/rowaggregation.cpp]: Do we need this commented line?

File: dbcon/joblist/jsonarrayagg.cpp [hunk 3]
```diff
@@ -0,0 +1,1079 @@
...
+
+        break;
+      }
+
+      case CalpontSystemCatalog::CHAR:
+      case CalpontSystemCatalog::VARCHAR:
+      case CalpontSystemCatalog::TEXT:
+      {
+        std::string maybeJson = row.getStringField(*i);
+        try
```
Comment (drrtuy) [inline dbcon/joblist/jsonarrayagg.cpp]: This try-catch doesn't look right and must be changed. It takes a lot to process every catch b/c it is a combination of internal interrupt + catch block processing.
There should be no try-catch that allows to go on processing in a low-level processing loop code. And outputRow is called for every record.

**Review by drrtuy — CHANGES_REQUESTED** [review]
Plz consider inherit JsonArrayInfo from GroupConcatInfo.

File: utils/rowgroup/rowaggregation.cpp (hunk 2 above)
Comment (mariadb-AndreyPiskunov) [reply utils/rowgroup/rowaggregation.cpp]: Will remove it and several other of my forgotten comments.

File: dbcon/joblist/jsonarrayagg.cpp (hunk 3 above)
Comment (mariadb-AndreyPiskunov) [reply dbcon/joblist/jsonarrayagg.cpp]: Fortunately, I found a flag for this function to not throw exceptions. Thanks.

File: dbcon/joblist/jsonarrayagg.h (hunk 1 above)
Comment (mariadb-AndreyPiskunov) [reply dbcon/joblist/jsonarrayagg.h]: Sorry, I always forget what year it is)

File: dbcon/mysql/ha_mcs_execplan.cpp [hunk 4]
```diff
@@ -2904,6 +2905,14 @@ uint32_t setAggOp(AggregateColumn* ac, Item_sum* isp)
       return rc;
     }
 
+    case Item_sum::JSON_ARRAYAGG_FUNC:
+    {
+      Item_func_group_concat* gc = (Item_func_group_concat*)isp;
```
Comment (drrtuy) [inline dbcon/mysql/ha_mcs_execplan.cpp]: JSON_ARRAYAGG_FUNC uses Item_func_group_concat?

File: dbcon/mysql/ha_mcs_execplan.cpp [hunk 5]
```diff
@@ -4909,7 +4918,8 @@ ReturnedColumn* buildAggregateColumn(Item* item, gp_walk_info& gwi)
   // Argument_count() is the # of formal parms to the agg fcn. Columnstore
   // only supports 1 argument except UDAnF, COUNT(DISTINC) and GROUP_CONCAT
```
Comment (drrtuy) [inline dbcon/mysql/ha_mcs_execplan.cpp]: It is worth to align the comment text also.

**Review by drrtuy — APPROVED** [review]

**Comment by drrtuy** [comment]
There is one thing left, namely the MTR test that is introduced with the patch fails to pass b/c of a randomly sorted output. See the build for details.

**Comment by drrtuy** [comment]
Look for sorted_results stanza [here](https://dev.mysql.com/doc/dev/mysql-server/latest/PAGE_MYSQL_TEST_COMMANDS.html).

**Review by drrtuy — CHANGES_REQUESTED** [review]
Take a look at the last comment.

File: dbcon/mysql/ha_mcs_execplan.cpp (hunk 5 above)
Comment (mariadb-AndreyPiskunov) [reply dbcon/mysql/ha_mcs_execplan.cpp]: Done

File: dbcon/mysql/ha_mcs_execplan.cpp (hunk 4 above)
Comment (mariadb-AndreyPiskunov) [reply dbcon/mysql/ha_mcs_execplan.cpp]: My bad, missed it.

**Comment by drrtuy** [comment]
LGTM

**Review by drrtuy — APPROVED** [review]

**Pull request merged**